homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -0,0 +1,121 @@
+"""DataUpdateCoordinator for the Yale integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from yalesmartalarmclient.client import AuthenticationError, YaleSmartAlarmClient
+
+from homeassistant.config_entries import SOURCE_REAUTH, ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, LOGGER
+
+
+class YaleDataUpdateCoordinator(DataUpdateCoordinator):
+    """A Yale Data Update Coordinator."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize the Yale hub."""
+        self.entry = entry
```

Comment:
```suggestion
        self.entry: ConfigEntry = entry
```
This is what Martin was suggesting when he said type the attributes. You can do this for each of your attributes, including self.yale which would be `Optional[YaleSmartAlarmClient]`. In addition to not having to ignore typing checks by CI, depending on your IDE it also makes it easier for you to reference these attributes later

Reply:
Note that the `entry` attribute is already typed since we type the corresponding parameter in the signature.

Reply:
TIL thanks!